### ISSUE TITLE
Mutate answers in custom action type

### ISSUE THREAD

**Issue by StevenLangbroek** [amwmedia/plop#118]
I'm not sure if I'm going about this wrong maybe, but I'm trying to enrich my `answers` object to make my templates a little bit simpler. What I did is create a custom action type called `prepare`. It takes a `prepare` config prop, which mutates the `answers` object it receives (I `Object.assign` the result onto the original `answers`. I know it's not ideal but I'm just trying to make this work first). In the next step however, they're not available. Is this intended behavior?

Here's my custom action type:

```js
exports.prepare = (answers, config) => {
  const { prepare } = config;
  Object.assign(answers, prepare(answers));
  return `Data prepared: ${JSON.stringify(answers, null, 2)}`;
};
```

I see it log the enriched `answers` object correctly, but the new props aren't available in the `templateFilePath` of the next step.

**Comment by StevenLangbroek** [comment]
I'm just looking at the merged PR in `node-plop`, and the easiest solution for my use-case, which I think has general applicability, is to allow `data` to be either an `object`, or a `function` that is called with `data`, so a `thunk` sort of. Thoughts?

**Comment by amwmedia** [comment]
yeah, I think this is actually a defect that was introduced as called out in #117. I'm going to take a look at this soon, I'm pretty sure it can just be addressed with a bug fix. 

I also really like your idea to make the data prop optionally take a function 👍

**Comment by lucasowens** [comment]
Any updates on this fix?

**Comment by amwmedia** [comment]
this is fixed in [v2.1.0](https://github.com/amwmedia/plop/releases/tag/v2.1.0), released this morning.
